routing/pathfind.go

```diff
@@ -587,11 +588,10 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 			timeLockDelta = edge.TimeLockDelta
 		}
 
-		incomingCltv := toNodeDist.incomingCltv +
-			uint32(timeLockDelta)
+		incomingCltv := toNodeDist.incomingCltv + int32(timeLockDelta)
 
 		// Check that we are within our CLTV limit.
-		if incomingCltv > r.CltvLimit {
+		if uint32(incomingCltv-finalHtlcExpiry) > r.CltvLimit {
```

Comment:
Wouldn't this be a bit cleaner in the form of `incomingCltv > finalHtlcExpiry + r.CltvLimit`?

Reply:
maybe safer indeed. but we can't cast `r.CltvLimit` because sometimes maxuint32 is used as a value.

Reply:
fixed

Reply:
Ok, also not ok, because with the addition we wrap around. Left as it was, but added comment about safety of this comparison.

Reply:
is it needed? i think nothing can go wrong as it is?

Reply:
changed just in case